lib/streams/request-stream.js

```diff
@@ -55,23 +48,14 @@ function RequestStream(requestStream, options) {
 			}
 		}
 	});
-	if(requestStream.method !== 'GET') {
-		inputPassThrough.pipe(requestStream, { noConvert: true });
+	CompoundDuplex.call(this, options);
+	if(requestStream.method === 'GET') {
+		this._compoundWrapNoWritablePipe = true;
```

Comment:
This seems like a hack.  Continuing the above concept of a WrapperStream or WrapStream, this should be implemented by creating a ReadableWrapStream, WritableWrapStream, and DuplexWrapStream .  These should inherit from their respective parents, where only DuplexWrapStream inherits from CompoundDuplex.